java/fury-core/src/main/java/io/fury/serializer/Serializers.java

```diff
@@ -410,17 +414,60 @@ public Double read(MemoryBuffer buffer) {
     }
   }
 
-  public static final class StringBuilderSerializer extends Serializer<StringBuilder> {
-    private final StringSerializer stringSerializer;
+  static Tuple2<ToByteFunction, Function> builderCache;
+
+  private static synchronized Tuple2<ToByteFunction, Function> getBuilderFunc() {
+    if (builderCache == null) {
+      Function getValue =
+          (Function) Functions.makeGetterFunction(StringBuilder.class.getSuperclass(), "getValue");
+      if (Platform.JAVA_VERSION > 8) {
+        ToByteFunction getCoder =
+            (ToByteFunction)
+                Functions.makeGetterFunction(StringBuilder.class.getSuperclass(), "getCoder");
+        builderCache = Tuple2.of(getCoder, getValue);
+      } else {
+        builderCache = Tuple2.of(null, getValue);
+      }
+    }
+    return builderCache;
+  }
 
-    public StringBuilderSerializer(Fury fury) {
-      super(fury, StringBuilder.class);
+  public abstract static class AbstractStringBuilderSerializer<T> extends Serializer<T> {
+    protected final ToByteFunction getCoder;
+    protected final Function getValue;
+    protected final StringSerializer stringSerializer;
+
+    public AbstractStringBuilderSerializer(Fury fury, Class<T> type) {
+      super(fury, type);
+      Tuple2<ToByteFunction, Function> builderFunc = getBuilderFunc();
+      getCoder = builderFunc.f0;
+      getValue = builderFunc.f1;
       stringSerializer = new StringSerializer(fury);
     }
 
     @Override
-    public void write(MemoryBuffer buffer, StringBuilder value) {
-      stringSerializer.writeJavaString(buffer, value.toString());
+    public void write(MemoryBuffer buffer, T value) {
+      if (Platform.JAVA_VERSION > 8) {
+        byte coder = getCoder.applyAsByte(value);
+        byte[] v = (byte[]) getValue.apply(value);
+        buffer.writeByte(coder);
+        buffer.writeBytesWithSizeEmbedded(v);
+      } else {
+        char[] v = (char[]) getValue.apply(value);
+        if (StringSerializer.isAscii(v)) {
+          stringSerializer.writeJDK8Ascii(buffer, v);
```

Comment:
```suggestion
          stringSerializer.writeJDK8Ascii(buffer, v, v.lengh());
```
Can we pass length to reduce extra data transfer?

Reply:
Yup,I to modify